### PR TITLE
SEAWEEDFS-K8S: helm chart

### PR DESCRIPTION
**Pull request by LazyDBA247-Anyvision**
seaweedfs on k8s deployment through helm chart & stateful sets

### CODE REVIEW

File: k8s/seaweedfs/values.yaml [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+  repository: null
+  imageName: null
+  imageTag: null
+  restartPolicy: null
+  replicas: 1
+  port: 8333
+  loggingOverrideLevel: null
+
+  # Folder on filer to store all buckets (default "/buckets")
+  filerDirBuckets: "/buckets"
```
Comment (chrislusf) [inline k8s/seaweedfs/values.yaml:277]: FYI: in 1.58, this setting is moved to filer.toml.

Comment (LazyDBA247-Anyvision) [reply k8s/seaweedfs/values.yaml:277]: correct, i already fixed, on "original" missed it on "your" white-labeled :)

**Comment by chrislusf** [comment]
Any instruction on how to run this? I tried the following.

```
k8s$ helm install --debug --dry-run seaweedfs ./seaweedfs
install.go:149: [debug] Original chart version: ""
install.go:166: [debug] CHART PATH: /Users/chris/go/src/github.com/chrislusf/seaweedfs/k8s/seaweedfs

Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(Deployment.spec): unknown field "serviceName" in io.k8s.api.apps.v1.DeploymentSpec
helm.go:75: [debug] error validating "": error validating data: ValidationError(Deployment.spec): unknown field "serviceName" in io.k8s.api.apps.v1.DeploymentSpec
helm.sh/helm/v3/pkg/kube.scrubValidationError
	/private/tmp/helm-20200130-53806-4ed745/src/helm.sh/helm/pkg/kube/client.go:539
helm.sh/helm/v3/pkg/kube.(*Client).Build
	/private/tmp/helm-20200130-53806-4ed745/src/helm.sh/helm/pkg/kube/client.go:136
helm.sh/helm/v3/pkg/action.(*Install).Run
	/private/tmp/helm-20200130-53806-4ed745/src/helm.sh/helm/pkg/action/install.go:229
main.runInstall
	/private/tmp/helm-20200130-53806-4ed745/src/helm.sh/helm/cmd/helm/install.go:214
main.newInstallCmd.func1
	/private/tmp/helm-20200130-53806-4ed745/src/helm.sh/helm/cmd/helm/install.go:116
github.com/spf13/cobra.(*Command).execute
	/private/tmp/helm-20200130-53806-4ed745/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826
github.com/spf13/cobra.(*Command).ExecuteC
	/private/tmp/helm-20200130-53806-4ed745/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914
github.com/spf13/cobra.(*Command).Execute
	/private/tmp/helm-20200130-53806-4ed745/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main
	/private/tmp/helm-20200130-53806-4ed745/src/helm.sh/helm/cmd/helm/helm.go:74
runtime.main
	/usr/local/Cellar/go/1.13.7/libexec/src/runtime/proc.go:203
runtime.goexit
	/usr/local/Cellar/go/1.13.7/libexec/src/runtime/asm_amd64.s:1357
unable to build kubernetes objects from release manifest
helm.sh/helm/v3/pkg/action.(*Install).Run
	/private/tmp/helm-20200130-53806-4ed745/src/helm.sh/helm/pkg/action/install.go:231
main.runInstall
	/private/tmp/helm-20200130-53806-4ed745/src/helm.sh/helm/cmd/helm/install.go:214
main.newInstallCmd.func1
	/private/tmp/helm-20200130-53806-4ed745/src/helm.sh/helm/cmd/helm/install.go:116
github.com/spf13/cobra.(*Command).execute
	/private/tmp/helm-20200130-53806-4ed745/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826
github.com/spf13/cobra.(*Command).ExecuteC
	/private/tmp/helm-20200130-53806-4ed745/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914
github.com/spf13/cobra.(*Command).Execute
	/private/tmp/helm-20200130-53806-4ed745/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main
	/private/tmp/helm-20200130-53806-4ed745/src/helm.sh/helm/cmd/helm/helm.go:74
runtime.main
	/usr/local/Cellar/go/1.13.7/libexec/src/runtime/proc.go:203
runtime.goexit
	/usr/local/Cellar/go/1.13.7/libexec/src/runtime/asm_amd64.s:1357
```

**Comment by LazyDBA247-Anyvision** [comment]
you are probably using helm 3.x?
the chart is for helm 2.x...

helm install --debug --dry-run --name=seaweedfs ./seaweedfs
on helm 2.x is working...

**Comment by LazyDBA247-Anyvision** [comment]
`yonin@instance-4:~$ helm install --debug --dry-run --name=seaweedfs ./seaweedfs
[debug] Created tunnel using local port: '32861'

[debug] SERVER: "127.0.0.1:32861"

[debug] Original chart version: ""
[debug] CHART PATH: /home/yonin/seaweedfs

NAME:   seaweedfs
REVISION: 1
RELEASED: Tue Mar  3 08:58:12 2020
CHART: seaweedfs-1.57
USER-SUPPLIED VALUES:
{}

COMPUTED VALUES:
certificates:
  commonName: SeaweedFS CA
  duration: 2160h
  ipAddresses: []
  keyAlgorithm: rsa
  keySize: 2048
  renewBefore: 360h
filer:
  affinity: |
    podAntiAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        - labelSelector:
.........
`

**Pull request merged**